src/subscription/azext_subscription/custom.py

```diff
@@ -3,16 +3,86 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
-from azext_subscription.subscription.models import SubscriptionDefinition
+from knack.log import get_logger
+from knack.util import CLIError
+from azext_subscription.subscription.models import (SubscriptionCreationParameters, AdPrincipal)
 
+logger = get_logger(__name__)
 
-def cli_subscription_create_subscription_definition(client, subscription_definition_name,
-                                                    offer_type, subscription_display_name=None):
-    if subscription_display_name is None:
-        subscription_display_name = subscription_definition_name
 
-    new_def = SubscriptionDefinition(
-        subscription_display_name=subscription_display_name,
-        offer_type=offer_type)
+def _get_object_id_by_spn(graph_client, spn):
+    accounts = list(graph_client.service_principals.list(
+        filter="servicePrincipalNames/any(c:c eq '{}')".format(spn)))
+    if not accounts:
+        logger.warning("Unable to find user with spn '%s'", spn)
+        return None
+    if len(accounts) > 1:
+        logger.warning("Multiple service principals found with spn '%s'. "
```

Comment:
`spn`/`upn` is supposed to be unique, I would suggest you just error out here as such situation should not occur

Reply:
This code is copied from KeyVault's policy handling. The parameter names (object_id/spn/upn) also reflect what both the KeyVault and AuthZ modules are doing. Ideally there'd be a separate change to address this feedback across all modules. What do you think?

Reply:
Sounds good to me